src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.cs

```diff
@@ -57,43 +65,85 @@ internal abstract partial class AbstractNavigateToSearchService : INavigateToSea
                 await client.TryInvokeAsync<IRemoteNavigateToSearchService>(
                     solution,
                     (service, solutionInfo, callbackId, cancellationToken) =>
-                    service.SearchDocumentAsync(solutionInfo, document.Id, searchPattern, kinds.ToImmutableArray(), callbackId, isFullyLoaded, cancellationToken),
+                    service.SearchDocumentAsync(solutionInfo, document.Id, searchPattern, kinds.ToImmutableArray(), callbackId, cancellationToken),
                     callback, cancellationToken).ConfigureAwait(false);
 
                 return;
             }
 
-            await SearchDocumentInCurrentProcessAsync(
-                document, searchPattern, kinds, onItemFound, isFullyLoaded, cancellationToken).ConfigureAwait(false);
+            await SearchFullyLoadedDocumentInCurrentProcessAsync(
+                document, searchPattern, kinds, onItemFound, cancellationToken).ConfigureAwait(false);
+        }
+
+        public Task SearchProjectAsync(Project project, ImmutableArray<Document> priorityDocuments, string searchPattern, IImmutableSet<string> kinds, Func<INavigateToSearchResult, Task> onResultFound, bool isFullyLoaded, CancellationToken cancellationToken)
+        {
+            return isFullyLoaded
```

Comment:
> But all projects may not be loaded yet so the document set being searched in the cached results may not be exhaustive.

Correct.  And the UI informs the user of that.  

> This probably isn't relevant to this PR specifically, but is it possible to cache the projects for a solution? So that even if the projects aren't fully loaded yet, the cached search can find them based on the solution being searched?

It is probably something we could do in teh future :)  it's def an interesting idea.  my general concern about those sorts of things is ensuring they get properly invalidated/updated as the solution changes over time.